### ISSUE TITLE
app crashes with interpolateMissingValues=false and areaChart=true

### ISSUE THREAD

**Issue by lajerca** [Abhinandan-Kushwaha/react-native-gifted-charts#805]
interpolateMissingValues={false} works well for simple line chart, but as soon as areaChart={true} the app crashes

### Error message
```text
java.lang.Error: Invalid number formating character 'e' (i=57, s=M0 110 L-1 34.682749999999984 L52.24675324675325 63.672 segmentStart{"startIndex":1,"endIndex":8,"color":"transparent"}segmentEndL105.4935064935065 110 L158.74025974025975 110 L211.987012987013 110 L265.23376623376623 110 L318.4805194805195 110 L371.72727272727275 110 L372.72727272727275 110 L0 110 )
  com.horcrux.svg.PathParser.parse_number(PathParser.java:627)
  com.horcrux.svg.PathParser.parse_list_number(PathParser.java:594)
  com.horcrux.svg.PathParser.parse(PathParser.java:166)
  com.horcrux.svg.PathView.setD(PathView.java:28)
  com.horcrux.svg.RenderableViewManager$PathViewManager.setD(RenderableViewManager.java:727)
  com.horcrux.svg.RenderableViewManager$PathViewManager.setD(RenderableViewManager.java:716)
  com.facebook.react.viewmanagers.RNSVGPathManagerDelegate.setProperty(RNSVGPathManagerDelegate.java:103)
  com.facebook.react.uimanager.ViewManagerPropertyUpdater.updateProps(ViewManagerPropertyUpdater.java:46)
  com.facebook.react.uimanager.ViewManager.updateProperties(ViewManager.java:77)
  com.facebook.react.uimanager.ViewManager.createViewInstance(ViewManager.java:181)
  com.facebook.react.uimanager.ViewManager.createView(ViewManager.java:108)
  com.facebook.react.uimanager.NativeViewHierarchyManager.createView(NativeViewHierarchyManager.java:281)
  com.facebook.react.uimanager.UIViewOperationQueue$CreateViewOperation.execute(UIViewOperationQueue.java:160)
  com.facebook.react.uimanager.UIViewOperationQueue$1.run(UIViewOperationQueue.java:904)
  com.facebook.react.uimanager.UIViewOperationQueue.flushPendingBatches(UIViewOperationQueue.java:1023)
  com.facebook.react.uimanager.UIViewOperationQueue.-$$Nest$mflushPendingBatches(Unknown Source:0)
  com.facebook.react.uimanager.UIViewOperationQueue$DispatchUIFrameCallback.doFrameGuarded(UIViewOperationQueue.java:1083)
  com.facebook.react.uimanager.GuardedFrameCallback.doFrame(GuardedFrameCallback.java:29)
  com.facebook.react.modules.core.ReactChoreographer$1.doFrame(ReactChoreographer.java:89)
  android.view.Choreographer$CallbackRecord.run(Choreographer.java:1337)
  android.view.Choreographer$CallbackRecord.run(Choreographer.java:1348)
  android.view.Choreographer.doCallbacks(Choreographer.java:952)
  android.view.Choreographer.doFrame(Choreographer.java:878)
  android.view.Choreographer$FrameDisplayEventReceiver.run(Choreographer.java:1322)
  android.os.Handler.handleCallback(Handler.java:958)
  android.os.Handler.dispatchMessage(Handler.java:99)
  android.os.Looper.loopOnce(Looper.java:205)
  android.os.Looper.loop(Looper.java:294)
  android.app.ActivityThread.main(ActivityThread.java:8177)
  java.lang.reflect.Method.invoke(Native Method)
  com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:552)
  com.android.internal.os.ZygoteInit.main(ZygoteInit.java:971)


```

**Comment by Abhinandan-Kushwaha** [comment]
Hi @lajerca If you set `interpolateMissingValues={false}` and your data has some missing values, then the chart will have breaks. It will not be a continuous line-chart. Area chart is possible only for continuous line charts, not for charts with breaks.

So `interpolateMissingValues={false}` and `areaChart={true}` is a crashing combination if there are some missing values. It is not a bug, but an expected behaviour.

**Comment by Abhinandan-Kushwaha** [comment]
Hi @lajerca Thanks for this suggestion. I have made a change that avoids crash, and just skips the area gradient if there are missing values and `interpolateMissingValues` is set to false.

It is available from versions `1.4.33` onwards. 🎉

Please use the latest version of the library.
